### PR TITLE
Not to call setState after dispose.

### PR DESCRIPTION
**Pull request by uhm0311**
Hello. I'm new to github and flutter. 
After I've used Flutter Staggered Animations, I found that if AnimationLimiter is in the TabBarView, It can cause error like below.

```
════════ Exception caught by scheduler library ═════════════════════════════════════════════════════
The following assertion was thrown during a scheduler callback:
setState() called after dispose(): _AnimationLimiterState#7d53d(lifecycle state: defunct, not mounted)

This error happens if you call setState() on a State object for a widget that no longer appears in the widget tree (e.g., whose parent widget no longer includes the widget in its build). This error can occur when code calls setState() from a timer or an animation callback.

The preferred solution is to cancel the timer or stop listening to the animation in the dispose() callback. Another solution is to check the "mounted" property of this object before calling setState() to ensure the object is still in the tree.
This error might indicate a memory leak if setState() is being called because another object is retaining a reference to this State object after it has been removed from the tree. To avoid memory leaks, consider breaking the reference to this object during dispose().

When the exception was thrown, this was the stack: 
#0      State.setState.<anonymous closure> (package:flutter/src/widgets/framework.dart:1112:9)
#1      State.setState (package:flutter/src/widgets/framework.dart:1147:6)
#2      _AnimationLimiterState.initState.<anonymous closure> (package:flutter_staggered_animations/src/animation_limiter.dart:45:7)
#3      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1102:15)
#4      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1049:9)
...
════════════════════════════════════════════════════════════════════════════════════════════════════
```

Please consider merge it.
Thanks.

### CODE REVIEW

File: lib/src/animation_limiter.dart [hunk 1]
```diff
@@ -36,15 +36,16 @@ class AnimationLimiter extends StatefulWidget {
 
 class _AnimationLimiterState extends State<AnimationLimiter> {
   bool _shouldRunAnimation = true;
+  bool _afterDispose = false;
```
Comment (uhm0311) [inline lib/src/animation_limiter.dart]: Used to whether if disposed or not.

File: lib/src/animation_limiter.dart [hunk 2]
```diff
@@ -36,15 +36,16 @@ class AnimationLimiter extends StatefulWidget {
...
   void initState() {
     super.initState();
 
     WidgetsBinding.instance.addPostFrameCallback((Duration value) {
-      setState(() {
-        _shouldRunAnimation = false;
-      });
+      if (!_afterDispose && this.mounted) {
+        setState(() => _shouldRunAnimation = false);
+      }
```
Comment (uhm0311) [inline lib/src/animation_limiter.dart]: Not to call setState after dispose

File: lib/src/animation_limiter.dart [hunk 3]
```diff
@@ -55,6 +56,12 @@ class _AnimationLimiterState extends State<AnimationLimiter> {
       child: widget.child,
     );
   }
+
+  @override
+  void dispose() {
+    _afterDispose = true;
```
Comment (uhm0311) [inline lib/src/animation_limiter.dart]: Check if disposed or not

**Comment by NearHuscarl** [comment]
Look good to me. Hope this PR will be merged.

**Comment by kw2019ltd** [comment]
when this PR will merged i can see errors in my logs regarding this issue

**Comment by MrLepage** [comment]
Duplicate of #31 
Will be fixed in 0.1.3

**Pull request closed without merging**